CMakeLists.txt

```diff
@@ -1,84 +1,99 @@
 
-# Find site-packages directory (needed for tests)
-execute_process(
-    COMMAND ${Python_EXECUTABLE} -c "import site; print(site.getsitepackages()[0])"
-    OUTPUT_VARIABLE PYTHON_SITE_PACKAGES_DIRECTORY
-    OUTPUT_STRIP_TRAILING_WHITESPACE
+#
+# Requirements: python3, opencv
+#
+find_package(Python3 REQUIRED COMPONENTS Development Interpreter)
+find_package(OpenCV REQUIRED)
+
+#
+# Define the main library
+#
+file(GLOB sources src/*.cpp)
+file(GLOB headers include/cvnp/*.h)
+add_library(cvnp ${sources})
+set_target_properties(cvnp PROPERTIES POSITION_INDEPENDENT_CODE ON)
+target_link_libraries(cvnp PUBLIC pybind11::pybind11 ${OpenCV_LIBS} Python3::Python)
+target_include_directories(
+    cvnp
+    PUBLIC
+    ${CMAKE_SOURCE_DIR}/include
+    ${OpenCV_INCLUDE_DIRS}
+)
+set_target_properties(cvnp PROPERTIES PUBLIC_HEADER "${headers}")
+install(TARGETS cvnp
+    LIBRARY DESTINATION lib
+    PUBLIC_HEADER DESTINATION include/cvnp
 )
 
+if(CVNP_BUILD_TESTS)
+    if(NOT Python_EXECUTABLE)
+        message(FATAL_ERROR "Could not find a python executable")
+    endif()
 
-if (PROJECT_IS_TOP_LEVEL)
-    # Add address sanitizer
-    option(CVNP_ENABLE_ASAN "Enable Address Sanitizer" OFF)
-    if (CVNP_ENABLE_ASAN)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address")
-        add_compile_definitions(CVNP_ENABLE_ASAN)
+    execute_process(
+        COMMAND ${Python_EXECUTABLE} -c "import numpy"
+        RESULT_VARIABLE NUMPY_NOT_FOUND
+    )
+
+    if(NOT ${NUMPY_NOT_FOUND} EQUAL 0)
+        message(FATAL_ERROR "numpy must be installed. Interpreter used '${Python_EXECUTABLE}'")
     endif()
 
-    #
-    # Requirements: python3, opencv
-    #
-    find_package(Python3 REQUIRED COMPONENTS Development Interpreter)
-    find_package(OpenCV REQUIRED)
+    # Find site-packages directory (needed for tests)
+    execute_process(
+        COMMAND ${Python_EXECUTABLE} -c "import site; print(site.getsitepackages()[0])"
+        OUTPUT_VARIABLE PYTHON_SITE_PACKAGES_DIRECTORY
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
 
-    # Requirement: pybind11
-    # Add pybind11 via add_subdirectory (since vcpkg's pybind11 fails on linux + python 3.12)
-    if (NOT IS_DIRECTORY(${CMAKE_CURRENT_LIST_DIR}/pybind11))
-        message(STATUS "cloning pybind11")
-        execute_process(
-            COMMAND git clone https://github.com/pybind/pybind11.git
-            WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
-        )
+    if (CVNP_ENABLE_ASAN)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address")
+        add_compile_definitions(CVNP_ENABLE_ASAN)
     endif()
-    add_subdirectory(pybind11)
 
     #
     # Build cvnp python module
     #
-    file(GLOB sources cvnp/*.h cvnp/*.cpp)
-    pybind11_add_module(cvnp ${sources} main/cvnp_main.cpp)
-    target_link_libraries(cvnp PRIVATE opencv_core)
-    target_include_directories(cvnp PUBLIC ${CMAKE_CURRENT_LIST_DIR})
     # Copy the .so or .pyd module into this dir, to facilitate the tests
```

Comment:
This comment ("copy the .so or .pyd module") was dealing with a part of code you removed (`add_custom_command`). Either remove the comment, or preferably re-add the `add_custom_command`.